ros2topic/ros2topic/verb/echo.py

```diff
@@ -95,16 +97,20 @@ def main(args):
         callback = subscriber_cb_csv(truncate_length, args.no_arr, args.no_str)
     qos_profile = qos_profile_from_short_keys(
         args.qos_profile, reliability=args.qos_reliability, durability=args.qos_durability)
+    if args.message_type is None:
+        with NodeStrategy(args) as node:
```

Comment:
This potentially creates a second direct node (beside the one created a few lines below) which doesn't seem like a good idea.

Same pattern below in multiple cases (even though they preexisted before).

Reply:
That is true, you might end up creating node, destroying it and creating another one. We do want to use the daemon if available though.

A solution could be to extend `NodeStrategy`'s API to ensure you're dealing with a direct node in case you have to. Something like a `force_direct()` method (or a context manager?). WDYT?

Reply:
> A solution could be to extend NodeStrategy's API to ensure you're dealing with a direct node in case you have to.

What is the difference to a `DirectNode` then?

Reply:
I think this should create direct node, and conditionally if a daemon is running query some information from that to not have to wait for "complete" discovery for the direct node.

Reply:
> What is the difference to a `DirectNode` then?

That it'd use the daemon outside the scope where you actually need it to be direct, e.g.:

```python
with NodeStrategy(args) as node:
    # ...
    with node.force_direct():
        node.subscribe()
```

> I think this should create direct node, and conditionally if a daemon is running query some information from that to not have to wait for "complete" discovery for the direct node.

That sounds a lot like replicating `NodeStrategy`'s implementation N times. I'd rather improve it in a single place.

Reply:
See https://github.com/ros2/ros2cli/pull/499 for one alternative.

Reply:
> See #499 for an alternative.

This looks much better :+1: